src/sliding-sync-sdk.ts

```diff
@@ -253,22 +253,48 @@ class ExtensionTyping implements Extension {
         };
     }
 
-    public onResponse(data: {rooms: Record<string, Event[]>}): void {
+    public onResponse(data: {rooms: Record<string, IMinimalEvent>}): void {
         if (!data || !data.rooms) {
             return;
         }
 
         for (const roomId in data.rooms) {
-            const ephemeralEvents = mapEvents(this.client, roomId, [data.rooms[roomId]]);
-            const room = this.client.getRoom(roomId);
-            if (!room) {
-                logger.warn("got typing events for room but room doesn't exist on client:", roomId);
-                continue;
-            }
-            room.addEphemeralEvents(ephemeralEvents);
-            ephemeralEvents.forEach((e) => {
-                this.client.emit(ClientEvent.Event, e);
-            });
+            processEphemeralEvents(
+                this.client, roomId, [data.rooms[roomId]],
+            );
+        }
+    }
+}
+
+class ExtensionReceipts implements Extension {
+    public constructor(private readonly client: MatrixClient) {}
+
+    public name(): string {
+        return "receipts";
+    }
+
+    public when(): ExtensionState {
+        return ExtensionState.PostProcess;
+    }
+
+    public onRequest(isInitial: boolean): object | undefined {
+        if (isInitial) {
+            return {
+                enabled: true,
+            };
+        }
+        return undefined;
```

Comment:
I wouldn't have minded a comment to explain why this logic is correct, but that could be due to unfamiliarity with sliding sync and the modules therein